gnomad/variant_qc/pipeline.py

```diff
@@ -89,31 +87,27 @@ def update_bin_expr(
         )
         return bin_expr
 
-    ht = ht.annotate(
-        singleton=ht.ac_raw == 1, snv=hl.is_snp(ht.alleles[0], ht.alleles[1]),
-    )
-
-    ht = ht.filter(ht.ac_raw > 0).persist()
-
     # Desired bins and sub-bins
     bin_expr = {"bin": True}
 
     if singleton:
-        bin_expr = update_bin_expr(bin_expr, ht.singleton, "singleton")
+        bin_expr = _update_bin_expr(bin_expr, ht.ac_raw == 1, "singleton")
 
     if biallelic:
-        bin_expr = update_bin_expr(bin_expr, ~ht.was_split, "biallelic")
+        bin_expr = _update_bin_expr(bin_expr, ~ht.was_split, "biallelic")
 
     if adj:
-        bin_expr = update_bin_expr(bin_expr, (ht.ac > 0), "adj")
+        bin_expr = _update_bin_expr(bin_expr, (ht.ac > 0), "adj")
 
     if add_substrat:
         for add_id, add_expr in add_substrat.items():
-            bin_expr = update_bin_expr(bin_expr, add_expr, add_id)
+            bin_expr = _update_bin_expr(bin_expr, add_expr, add_id)
 
     bin_ht = compute_quantile_bin(
         ht, score_expr=ht.score, bin_expr=bin_expr, n_bins=n_bins
     )
+
+    ht = ht.select_globals()
```

Comment:
basic question, why do you not want the globals?

Reply:
Because the function I pass this to modifies the HT and returns it while adding globals. When I do the join, I only want one copy of the original globals